ynr/apps/popolo/migrations/0034_populate_from_ee_divisions.py

```diff
@@ -0,0 +1,136 @@
+"""
+This data migration takes a CSV* exported from EveryElection
+and uses it to match up and change the identifier for each post it finds.
+
+When it's found a post, it makes a new `PostIdentifier` model with
+the previous `slug` value, updates the slug and sets other fields like
+start and end dates, etc.
+
+* It's actually two files to deal with two cases:
+
+1. The ballot has a post that is a division of an organisation
+2. The ballot has a post that's actually an organisation, but where
+   we created a pseudo-post for it at import time (for directly elected mayors,
+   PCCs, etc)
+
+"""
```

Comment:
I've attempted to do some kind of sanity checks on this but I'm conscious that I don't really understand how the YNR entities relate to each other to fully know where the murky corners of this are going to be.
One of the checks I looked at was for posts that aren't attached to any ballot ( `SELECT * FROM popolo_post WHERE id NOT IN (SELECT post_id FROM candidates_ballot)` ). Intuitively, I would have thought this process should be reducing that problem, but before applying this migration I had 411 and then afterwards I've got 1,293. Is that expected to you?

Reply:
Right, this is actually _a good thing_  – the different numbers are the posts with no ballot before and after. The reason the number grows after is because the ballots have been attached to the correct posts.

I've not attempted to clean up the old posts yet – I'm happy having them in the DB for the time being.

One thing I've just thought of though is that there are a few places that we still rely on the post<>membership relationships. I'll update the migration to move them over too